sockeye/data_io.py

```diff
@@ -874,25 +906,26 @@ class DataConfig(config.Config):
     """
 
     def __init__(self,
-                 source: str,
+                 sources: List[str],
```

Comment:
have you guys tried running a model trained with an old sockeye version with the source factor version? I suspect that it will fail loading the data config, which would kind of silly as the sources and and source_vocabs is mainly for documentation. If that's the case in the long term it probably would make sense to move everything not used at inference time out of DataConfig into either a dedicated file or just have it in the log.

Reply:
good point, we should check this.

Reply:
no longer relevant as this PR now bumps major version.